### ISSUE TITLE
execution halted

### ISSUE THREAD

**Issue by badar-chachar** [oscarpobletes/GitHubMines#2]
Hi there,
I have been using this program extensively. I get the following error repeatedly, is there any fix for it?

```
Error: Something went wrong while executing your query. This may be the result of a timeout, or it could be a GitHub bug. Please include `B532:514E:9D31833:9F4D8A8:652D0C6F` when reporting this issue.: {"response":{"data":null,"errors":[{"message":"Something went wrong while executing your query. This may be the result of a timeout, or it could be a GitHub bug. Please include `B532:514E:9D31833:9F4D8A8:652D0C6F` when reporting this issue."}],"status":502},"request":{"query":"query ($searchQuery: String!, $first: Int, $after: String) {\n  search(query: $searchQuery, type: REPOSITORY, first: $first, after: $after) {\n    nodes {\n      ... on Repository {\n        nameWithOwner\n        description\n        url\n        createdAt\n        assignableUsers {\n          totalCount\n        }\n        watchers {\n          totalCount\n        }\n        stargazerCount\n        projects {\n          totalCount\n        }\n        issues {\n          totalCount\n        }\n        pullRequests {\n          totalCount\n        }\n        diskUsage\n        licenseInfo {\n          spdxId\n        }\n        languages(first: 5) {\n          edges {\n            node {\n              name\n            }\n          }\n        }\n        primaryLanguage {\n          name\n        }\n        environments(first: 5) {\n          edges {\n            node {\n              name\n            }\n          }\n        }\n        submodules(first: 5) {\n          edges {\n            node {\n              name\n            }\n          }\n        }\n        repositoryTopics(first: 5) {\n          edges {\n            node {\n              topic {\n                name\n              }\n            }\n          }\n        }\n      }\n    }\n    pageInfo {\n      endCursor\n      hasNextPage\n    }\n  }\n}","variables":{"searchQuery":"query-string created:2021-01-06","first":50,"after":null}}}
    at GraphQLClient.<anonymous> (/home/bdr/src/GitHubMines/node_modules/graphql-request/dist/src/index.js:116:35)
    at step (/home/bdr/src/GitHubMines/node_modules/graphql-request/dist/src/index.js:40:23)
    at Object.next (/home/bdr/src/GitHubMines/node_modules/graphql-request/dist/src/index.js:21:53)
    at fulfilled (/home/bdr/src/GitHubMines/node_modules/graphql-request/dist/src/index.js:12:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  response: { data: null, errors: [ [Object] ], status: 502 },
  request: {
    query: 'query ($searchQuery: String!, $first: Int, $after: String) {\n' +
      '  search(query: $searchQuery, type: REPOSITORY, first: $first, after: $after) {\n' +
      '    nodes {\n' +
      '      ... on Repository {\n' +
      '        nameWithOwner\n' +
      '        description\n' +
      '        url\n' +
      '        createdAt\n' +
      '        assignableUsers {\n' +
      '          totalCount\n' +
      '        }\n' +
      '        watchers {\n' +
      '          totalCount\n' +
      '        }\n' +
      '        stargazerCount\n' +
      '        projects {\n' +
      '          totalCount\n' +
      '        }\n' +
      '        issues {\n' +
      '          totalCount\n' +
      '        }\n' +
      '        pullRequests {\n' +
      '          totalCount\n' +
      '        }\n' +
      '        diskUsage\n' +
      '        licenseInfo {\n' +
      '          spdxId\n' +
      '        }\n' +
      '        languages(first: 5) {\n' +
      '          edges {\n' +
      '            node {\n' +
      '              name\n' +
      '            }\n' +
      '          }\n' +
      '        }\n' +
      '        primaryLanguage {\n' +
      '          name\n' +
      '        }\n' +
      '        environments(first: 5) {\n' +
      '          edges {\n' +
      '            node {\n' +
      '              name\n' +
      '            }\n' +
      '          }\n' +
      '        }\n' +
      '        submodules(first: 5) {\n' +
      '          edges {\n' +
      '            node {\n' +
      '              name\n' +
      '            }\n' +
      '          }\n' +
      '        }\n' +
      '        repositoryTopics(first: 5) {\n' +
      '          edges {\n' +
      '            node {\n' +
      '              topic {\n' +
      '                name\n' +
      '              }\n' +
      '            }\n' +
      '          }\n' +
      '        }\n' +
      '      }\n' +
      '    }\n' +
      '    pageInfo {\n' +
      '      endCursor\n' +
      '      hasNextPage\n' +
      '    }\n' +
      '  }\n' +
      '}',
    variables: {
      searchQuery: 'query-string created:2021-01-06',
      first: 50,
      after: null
    }
  }
}
TypeError: result is not iterable (cannot read property undefined)
    at fetchAllResults (/home/bdr/src/GitHubMines/index.js:92:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
TypeError: Cannot read properties of undefined (reading 'map')
    at writeFiles (/home/bdr/src/GitHubMines/index.js:104:33)
    at /home/bdr/src/GitHubMines/index.js:290:5
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

```

**Comment by ksegla** [comment]
Hi Bachar
I will look into it today.

**Comment by ksegla** [comment]
@badar-chachar 
I think the fastest way to solve this is to set the batch size lower than 50 (try 10). This seems to be a GitHub problem. In my tests, it always works with 10 but sometimes fails with 50. It is likely due to the volume of information; some of the results are bigger and need too much time. It happens even in the GraphQL API Explorer from the GitHub site, depending on the results. 

Look in the code for 
`const batchSize = 50; // Set the desired batch size. Maximum is 100`
and set it to 10.
Let us know if it solved your issue. I will look into ways to keep a big batch size without running into this t

(There are also some possible improvements in the way we get the results. I will test and commit a more efficient version of the tool later today.)

**Comment by ksegla** [comment]
@badar-chachar 
You can find a newer version of this project here: 
https://github.com/ksegla/GitHubMiner
It allows for batchsize as a parameter and is more efficient (in terms of the date range splitting that we use for queries with more than 1000 results.)

**Comment by badar-chachar** [comment]
> @badar-chachar I think the fastest way to solve this is to set the batch size lower than 50 (try 10). This seems to be a GitHub problem. In my tests, it always works with 10 but sometimes fails with 50. It is likely due to the volume of information; some of the results are bigger and need too much time. It happens even in the GraphQL API Explorer from the GitHub site, depending on the results.
> 
> Look in the code for `const batchSize = 50; // Set the desired batch size. Maximum is 100` and set it to 10. Let us know if it solved your issue. I will look into ways to keep a big batch size without running into this t
> 
> (There are also some possible improvements in the way we get the results. I will test and commit a more efficient version of the tool later today.)

@ksegla I tried `batchSize = 10` and it works fine. I tried it for a query returning 60k results in one go. Before I had to split for year range manually.

**Comment by badar-chachar** [comment]
> @badar-chachar You can find a newer version of this project here: https://github.com/ksegla/GitHubMiner It allows for batchsize as a parameter and is more efficient (in terms of the date range splitting that we use for queries with more than 1000 results.)

Hello, i used this and it doesn't terminate no metter how long the result size is.

**Comment by ksegla** [comment]
@badar-chachar 
Thanks for the feedback. Could you provide me with an example of a query that doesn't terminate. It terminates in all my tests. 

(

For context, the newer version tries to be more efficient by scanning ranges of dates, instead of going one day at a time. To give you an extreme example, suppose you have a query for projects created between 2013-01-01 and 2023-01-01 and that the count of projects is 1001. The current version, in order to avoid the 1000 GitHub limit, would try every single day between those dates (around 3650 queries), which can be long and also consumes more of your rate limit. The newer version proceeds by iteratively splitting the date ranges in 2 and checking if there are more than 1000 results in the newer date ranges. So in an extreme scenario, to ensure you have less than 1000 results, you could need 3650+ queries for the first version but only 2 in the second version.

)

**Comment by badar-chachar** [comment]
@ksegla I got 66K results for `ethereum` and 14K for `cosmos`. I just passed these strings as query without other parameters of date range or batchsize and program didn't crash.

**Comment by ksegla** [comment]
@badar-chachar 
Ah, I didn't understand at first. I thought you meant that GitHubMiner, the new version, was looping indefinitely (which was plausible since the new version uses trickier stop conditions) but it seems like you meant that the new version is working fine. Please note that batchsize is at 10 by default in GitHubMiner, so that's probably why you didn't have to set it. There<s still an issue with a batch size of 50.

Also, I just did a commit that fixed a possible problem about the default values. If you didn't set a start date, be aware that the tool was previously picking 2013-01-01 as a default date.  I changed it to 2008-01-01, just before GitHub got released. 

For new issues, please use 
https://github.com/ksegla/GitHubMiner

Thanks

**Comment by badar-chachar** [comment]
@ksegla Thanks again, I just wanted to give feedback for the new version. I will put new issue on https://github.com/ksegla/GitHubMiner definitely. I will keep in mind the default date modification.
